IncompletePostTestCase fails once in a while with SocketTimeoutException: Read timed out

I've noticed the `IncompletePostTestCase` fail once in while in our CI jobs. One such instance is[1]. The failure looks like:

```
2020-09-02T09:19:26.0616166Z [INFO] Running io.quarkus.resteasy.test.IncompletePostTestCase
2020-09-02T09:19:27.5212202Z 2020-09-02 09:19:27,379 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT on JVM started in 0.726s. Listening on: http://0.0.0.0:8081
2020-09-02T09:19:27.5213544Z 2020-09-02 09:19:27,379 INFO  [io.quarkus] (main) Profile test activated. 
2020-09-02T09:19:27.5213930Z 2020-09-02 09:19:27,379 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, security]
2020-09-02T09:19:30.9691281Z 2020-09-02 09:19:30,934 INFO  [io.quarkus] (main) Quarkus stopped in 0.435s
2020-09-02T09:19:30.9757067Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 5.832 s <<< FAILURE! - in io.quarkus.resteasy.test.IncompletePostTestCase
2020-09-02T09:19:30.9759563Z [ERROR] io.quarkus.resteasy.test.IncompletePostTestCase.testIncompleteWrite  Time elapsed: 3.108 s  <<< ERROR!
2020-09-02T09:19:30.9759882Z java.net.SocketTimeoutException: Read timed out
2020-09-02T09:19:30.9760137Z 	at java.base/java.net.SocketInputStream.socketRead0(Native Method)
2020-09-02T09:19:30.9760406Z 	at java.base/java.net.SocketInputStream.socketRead(SocketInputStream.java:115)
2020-09-02T09:19:30.9760721Z 	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:168)
2020-09-02T09:19:30.9760982Z 	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:140)
2020-09-02T09:19:30.9761278Z 	at org.apache.http.impl.io.AbstractSessionInputBuffer.fillBuffer(AbstractSessionInputBuffer.java:161)
2020-09-02T09:19:30.9761573Z 	at org.apache.http.impl.io.SocketInputBuffer.fillBuffer(SocketInputBuffer.java:82)
2020-09-02T09:19:30.9761892Z 	at org.apache.http.impl.io.AbstractSessionInputBuffer.readLine(AbstractSessionInputBuffer.java:276)
2020-09-02T09:19:30.9762176Z 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
2020-09-02T09:19:30.9762457Z 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
2020-09-02T09:19:30.9762727Z 	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
2020-09-02T09:19:30.9763011Z 	at org.apache.http.impl.AbstractHttpClientConnection.receiveResponseHeader(AbstractHttpClientConnection.java:294)
2020-09-02T09:19:30.9763299Z 	at org.apache.http.impl.conn.DefaultClientConnection.receiveResponseHeader(DefaultClientConnection.java:257)
2020-09-02T09:19:30.9763589Z 	at org.apache.http.impl.conn.ManagedClientConnectionImpl.receiveResponseHeader(ManagedClientConnectionImpl.java:207)
2020-09-02T09:19:30.9763880Z 	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
2020-09-02T09:19:30.9764151Z 	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
2020-09-02T09:19:30.9764728Z 	at org.apache.http.impl.client.DefaultRequestDirector.tryExecute(DefaultRequestDirector.java:679)
2020-09-02T09:19:30.9764985Z 	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:481)
2020-09-02T09:19:30.9765259Z 	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
2020-09-02T09:19:30.9765529Z 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
2020-09-02T09:19:30.9765801Z 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
2020-09-02T09:19:30.9766060Z 	at org.apache.http.client.HttpClient$execute$0.call(Unknown Source)
2020-09-02T09:19:30.9767450Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
2020-09-02T09:19:30.9767827Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
2020-09-02T09:19:30.9768102Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
2020-09-02T09:19:30.9768387Z 	at io.restassured.internal.RequestSpecificationImpl$RestAssuredHttpBuilder.doRequest(RequestSpecificationImpl.groovy:2055)
2020-09-02T09:19:30.9768666Z 	at io.restassured.internal.http.HTTPBuilder.doRequest(HTTPBuilder.java:495)
2020-09-02T09:19:30.9768959Z 	at io.restassured.internal.http.HTTPBuilder.request(HTTPBuilder.java:452)
2020-09-02T09:19:30.9769274Z 	at io.restassured.internal.http.HTTPBuilder$request$2.call(Unknown Source)
2020-09-02T09:19:30.9769545Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
2020-09-02T09:19:30.9769822Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
2020-09-02T09:19:30.9770092Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:166)
2020-09-02T09:19:30.9770529Z 	at io.restassured.internal.RequestSpecificationImpl.sendHttpRequest(RequestSpecificationImpl.groovy:1451)
2020-09-02T09:19:30.9770865Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-09-02T09:19:30.9771145Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-09-02T09:19:30.9772200Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-09-02T09:19:30.9772473Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-09-02T09:19:30.9772729Z 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
2020-09-02T09:19:30.9772989Z 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
2020-09-02T09:19:30.9773243Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1262)
2020-09-02T09:19:30.9773505Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1029)
2020-09-02T09:19:30.9773758Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:816)
2020-09-02T09:19:30.9774005Z 	at groovy.lang.GroovyObject.invokeMethod(GroovyObject.java:39)
2020-09-02T09:19:30.9852749Z 	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:45)
2020-09-02T09:19:30.9853325Z 	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.callCurrent(PogoInterceptableSite.java:55)
2020-09-02T09:19:30.9853616Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
2020-09-02T09:19:30.9853892Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
2020-09-02T09:19:30.9854171Z 	at io.restassured.internal.RequestSpecificationImpl.sendRequest(RequestSpecificationImpl.groovy:1200)
2020-09-02T09:19:30.9854445Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-09-02T09:19:30.9854723Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-09-02T09:19:30.9855010Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-09-02T09:19:30.9855279Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-09-02T09:19:30.9855537Z 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
2020-09-02T09:19:30.9855793Z 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
2020-09-02T09:19:30.9856045Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1262)
2020-09-02T09:19:30.9856298Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1029)
2020-09-02T09:19:30.9856569Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:816)
2020-09-02T09:19:30.9856820Z 	at groovy.lang.GroovyObject.invokeMethod(GroovyObject.java:39)
2020-09-02T09:19:30.9857251Z 	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:45)
2020-09-02T09:19:30.9860451Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
2020-09-02T09:19:30.9861582Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
2020-09-02T09:19:30.9861933Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:166)
2020-09-02T09:19:30.9862216Z 	at io.restassured.internal.filter.SendRequestFilter.filter(SendRequestFilter.groovy:30)
2020-09-02T09:19:30.9862479Z 	at io.restassured.filter.Filter$filter$0.call(Unknown Source)
2020-09-02T09:19:30.9862736Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
2020-09-02T09:19:30.9862990Z 	at io.restassured.filter.Filter$filter.call(Unknown Source)
2020-09-02T09:19:30.9863247Z 	at io.restassured.internal.filter.FilterContextImpl.next(FilterContextImpl.groovy:72)
2020-09-02T09:19:30.9863519Z 	at io.restassured.filter.time.TimingFilter.filter(TimingFilter.java:56)
2020-09-02T09:19:30.9863758Z 	at io.restassured.filter.Filter$filter.call(Unknown Source)
2020-09-02T09:19:30.9864014Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
2020-09-02T09:19:30.9864271Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
2020-09-02T09:19:30.9864532Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:157)
2020-09-02T09:19:30.9864792Z 	at io.restassured.internal.filter.FilterContextImpl.next(FilterContextImpl.groovy:72)
2020-09-02T09:19:30.9865045Z 	at io.restassured.filter.FilterContext$next.call(Unknown Source)
2020-09-02T09:19:30.9865299Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
2020-09-02T09:19:30.9865556Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
2020-09-02T09:19:30.9866301Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
2020-09-02T09:19:30.9867274Z 	at io.restassured.internal.RequestSpecificationImpl.applyPathParamsAndSendRequest(RequestSpecificationImpl.groovy:1655)
2020-09-02T09:19:30.9868078Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-09-02T09:19:30.9868373Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-09-02T09:19:30.9869542Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-09-02T09:19:30.9869835Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-09-02T09:19:30.9871635Z 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
2020-09-02T09:19:30.9871984Z 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
2020-09-02T09:19:30.9872238Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1262)
2020-09-02T09:19:30.9872510Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1029)
2020-09-02T09:19:30.9872764Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:816)
2020-09-02T09:19:30.9873016Z 	at groovy.lang.GroovyObject.invokeMethod(GroovyObject.java:39)
2020-09-02T09:19:30.9873279Z 	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:45)
2020-09-02T09:19:30.9873901Z 	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.callCurrent(PogoInterceptableSite.java:55)
2020-09-02T09:19:30.9874224Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
2020-09-02T09:19:30.9957514Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
2020-09-02T09:19:30.9957945Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)
2020-09-02T09:19:30.9958249Z 	at io.restassured.internal.RequestSpecificationImpl.applyPathParamsAndSendRequest(RequestSpecificationImpl.groovy:1661)
2020-09-02T09:19:30.9958795Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-09-02T09:19:30.9959127Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-09-02T09:19:30.9959411Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-09-02T09:19:30.9959675Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-09-02T09:19:30.9959933Z 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
2020-09-02T09:19:30.9960196Z 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
2020-09-02T09:19:30.9960448Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1262)
2020-09-02T09:19:30.9960706Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1029)
2020-09-02T09:19:30.9961133Z 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:816)
2020-09-02T09:19:30.9961448Z 	at groovy.lang.GroovyObject.invokeMethod(GroovyObject.java:39)
2020-09-02T09:19:30.9961719Z 	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:45)
2020-09-02T09:19:30.9961999Z 	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.callCurrent(PogoInterceptableSite.java:55)
2020-09-02T09:19:30.9962269Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
2020-09-02T09:19:30.9962538Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
2020-09-02T09:19:30.9962805Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)
2020-09-02T09:19:30.9963647Z 	at io.restassured.internal.RequestSpecificationImpl.get(RequestSpecificationImpl.groovy:171)
2020-09-02T09:19:30.9964074Z 	at io.restassured.internal.RequestSpecificationImpl.get(RequestSpecificationImpl.groovy)
2020-09-02T09:19:30.9964358Z 	at io.quarkus.resteasy.test.IncompletePostTestCase.testIncompleteWrite(IncompletePostTestCase.java:52)

```
I initially thought the timeouts configured on that test are slightly on the lower side. But it's been configured for a 1 second timeout, which I think is reasonable. It may very well be the network communication on the GitHub action VMs is slow and the timeout might need an increase, but before doing that we might need to look a bit into it. Given the context around which that test case was introduced[2], I don't want to blindly increase that timeout to get past it. 

[1] https://github.com/quarkusio/quarkus/pull/10769/checks?check_run_id=1060578412
[2] https://github.com/quarkusio/quarkus/pull/7674

--- a/extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/IncompletePostTestCase.java
+++ b/extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/IncompletePostTestCase.java
@@ -46,8 +46,8 @@ public class IncompletePostTestCase {
         Assertions.assertFalse(PostEndpoint.invoked);
         RestAssuredConfig config = RestAssured.config()
                 .httpClient(HttpClientConfig.httpClientConfig()
-                        .setParam(CoreConnectionPNames.CONNECTION_TIMEOUT, 1000)
-                        .setParam(CoreConnectionPNames.SO_TIMEOUT, 1000));
+                        .setParam(CoreConnectionPNames.CONNECTION_TIMEOUT, 60000)
+                        .setParam(CoreConnectionPNames.SO_TIMEOUT, 60000));
 
         RestAssured.given().config(config).get("/post").then().body(Matchers.is("ok"));
     }

